flo-workflow/src/main/java/com/spotify/flo/TaskOperator.java

```diff
@@ -0,0 +1,41 @@
+/*-
+ * -\-\-
+ * Flo Workflow Definition
+ * --
+ * Copyright (C) 2016 - 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.flo;
+
+public abstract class TaskOperator<T> extends TaskContextGeneric<T> {
```

Comment:
Could the job spec type `T` be restricted to some interface that better documents the expected behavior of throwing a `SpecException` on some `success` method?

Reply:
Would prefer to come up with a better mechanism instead of abusing exceptions.

Reply:
It's a bit ucky, but it's not really exposed to the user so I'm ready to let it slip. Will of course prefer a better solution. Will think a bit about it.

Reply:
I think I might retreat a bit and just run the operator directly in the process fn instead.

I realized that executing the operator outside the task process fn does not work well with our current forking strategy as the operator ends up executing on the main process instead of the task process. That might be fixable as well by introducing additional EvalContext methods to allow execution of arbitrary functions to be forked, but feels like the scope of this PR might be becoming too large.